### ISSUE TITLE
[Feature Request] Provide auto completion for Swift Language.

### ISSUE THREAD

**Issue by rahul1994joshi** [geany/geany#1078]
Please add the support for the swift language auto completion functionality .

**Comment by codebrainz** [comment]
The main part of this will be adding a parser for Swift to [ctags](https://github.com/universal-ctags/ctags), once that's done, it should be relatively trivial to add a new filetype for it. Scintilla's C++ lexer [apparently supports it to some extent](https://sourceforge.net/p/scintilla/scite/ci/fd45392dd14c79be7621227a5e998a274a5322c1/) already, for syntax highlighting.

**Comment by ankitpati** [comment]
PR #1265 may help with this. Please consider reviewing and merging it.

**Comment by b4n** [comment]
@ankitpati that PR won't help for auto-completion as it doesn't set a `tag_parser` filteype.

**Comment by elextr** [comment]
#1265 is a start, but it won't add autocompletion, as @codebrainz said above, that needs a tags parser.

**Comment by ankitpati** [comment]
I agree it will not help with autocompletion, but it gets the highlighting and compile/run/build shortcuts working.

**Comment by techee** [comment]
https://github.com/techee/geany-lsp together with https://github.com/swiftlang/sourcekit-lsp work based on my testing.
